src/Parser/GuardianDruid/Modules/Features/AntiFillerSpam.js

```diff
@@ -0,0 +1,199 @@
+      if (`${spellID}` === gcdSpellID) {
+        return;
+      }
+
+      const { isFiller, baseCD, hastedCD, condition, proc, isUtility } = GCD_SPELLS[gcdSpellID];
+      const lastCast = this.abilityLastCasts[gcdSpellID] || -Infinity;
+      const isFillerEval = (typeof isFiller === 'function') ? isFiller(event, combatant, targets, lastCast) : isFiller;
+      if (isFillerEval || isUtility) {
+        return;
+      }
+
+      let isOffCooldown = true;
+      if (baseCD !== null) {
+        if (hastedCD) {
+          isOffCooldown = this.dynamicHaste.isHastedAbilityOffCooldown(lastCast, timestamp, baseCD);
+        } else {
+          isOffCooldown = timestamp - lastCast > baseCD;
+        }
+      }
+
+      const hasProc = proc && ((typeof proc === 'function') ? proc(event, combatant, targets, lastCast) : proc);
+      const meetsCondition = (condition !== undefined) ? ((typeof condition === 'function') ? condition(event, combatant, targets, lastCast) : condition) : true;
+
+
+      if ((isOffCooldown || hasProc) && meetsCondition) {
+        debug && console.warn(`[Available non-filler] - ${gcdSpellID} ${SPELLS[gcdSpellID].name} - offCD: ${isOffCooldown} - hasProc: ${hasProc} - canBeCast: ${meetsCondition}`);
+        availableSpells.push(gcdSpellID);
+      }
+    });
+
+    if (availableSpells.length > 0) {
+      this._unnecessaryFillerSpells += 1;
+    }
+    debug && console.groupEnd();
+  }
+
+  get fillerSpamPercentage() {
+    return this._unnecessaryFillerSpells / this._totalGCDSpells;
+  }
+
+  statistic() {
+    return (
+      <StatisticBox
+        icon={<SpellIcon id={SPELLS.BEAR_SWIPE.id} />}
+        value={`${formatPercentage(this.fillerSpamPercentage)}%`}
+        label='Unnecessary Fillers'
+        tooltip={`You cast <strong>${this._unnecessaryFillerSpells}</strong> unnecessary filler spells out of <strong>${this._totalGCDSpells}</strong> total GCDs.  Filler spells (Swipe, Moonfire without a GG proc, or Moonfire outside of pandemic if talented into Incarnation) do far less damage than your main rotational spells, and should be minimized whenever possible.`}
+      />
+    );
+  }
+
+  suggestions(when) {
+    when(this.fillerSpamPercentage).isGreaterThan(0.1)
+      .addSuggestion((suggest, actual, recommended) => {
+        return suggest(<span>You are casting too many unnecessary filler spells. Try to plan your casts two or three GCDs ahead of time to anticipate your main rotational spells coming off cooldown, and to give yourself time to react to <SpellLink id={SPELLS.GORE_BEAR.id} /> and <SpellLink id={SPELLS.GALACTIC_GUARDIAN_TALENT.id} /> procs.</span>)
+          .icon(SPELLS.BEAR_SWIPE.icon)
+          .actual(`${formatPercentage(actual)}% unnecessary filler spells cast`)
+          .recommended(`${formatPercentage(recommended, 0)}% or less is recommended`)
+          .regular(recommended + 0.05).major(recommended + 0.1);
+      });
+  }
+}
+
+export default AntiFillerSpam;
+
+const GCD_SPELLS = {
```

Comment:
please define this prior to using it, better would be as a static property of the class so children can override it.

Reply:
I think another improvement would be to move this to the CastEfficiency abilities property.